{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "#headers\n",
    "# importing KEGG API\n",
    "import Bio\n",
    "Bio.__version__\n",
    "from Bio import SeqIO\n",
    "from Bio.KEGG.REST import *\n",
    "from Bio.KEGG.KGML import KGML_parser\n",
    "from Bio.KEGG import REST\n",
    "\n",
    "from IPython.display import Image, HTML\n",
    "\n",
    "\n",
    "import random\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Enter a Species ( mouse is mmu)mmu\n",
      "Enter a Gene: COx7b2\n",
      "ENTRY       78174             CDS       T01002\n",
      "NAME        Cox7b2, 4930503B16Rik\n",
      "DEFINITION  (RefSeq) cytochrome c oxidase subunit VIIb2\n",
      "ORTHOLOGY   K02271  cytochrome c oxidase subunit 7b\n",
      "ORGANISM    mmu  Mus musculus (mouse)\n",
      "PATHWAY     mmu00190  Oxidative phosphorylation\n",
      "            mmu01100  Metabolic pathways\n",
      "            mmu04260  Cardiac muscle contraction\n",
      "            mmu04714  Thermogenesis\n",
      "            mmu04932  Non-alcoholic fatty liver disease (NAFLD)\n",
      "            mmu05010  Alzheimer's disease\n",
      "            mmu05012  Parkinson's disease\n",
      "            mmu05016  Huntington's disease\n",
      "MODULE      mmu_M00154  Cytochrome c oxidase\n",
      "BRITE       KEGG Orthology (KO) [BR:mmu00001]\n",
      "             Metabolism\n",
      "              Energy metabolism\n",
      "               00190 Oxidative phosphorylation\n",
      "                78174 (Cox7b2)\n",
      "             Organismal Systems\n",
      "              Circulatory system\n",
      "               04260 Cardiac muscle contraction\n",
      "                78174 (Cox7b2)\n",
      "              Environmental adaptation\n",
      "               04714 Thermogenesis\n",
      "                78174 (Cox7b2)\n",
      "             Human Diseases\n",
      "              Neurodegenerative diseases\n",
      "               05010 Alzheimer's disease\n",
      "                78174 (Cox7b2)\n",
      "               05012 Parkinson's disease\n",
      "                78174 (Cox7b2)\n",
      "               05016 Huntington's disease\n",
      "                78174 (Cox7b2)\n",
      "              Endocrine and metabolic diseases\n",
      "               04932 Non-alcoholic fatty liver disease (NAFLD)\n",
      "                78174 (Cox7b2)\n",
      "POSITION    5; 5 C3.1\n",
      "MOTIF       Pfam: COX7B KdpA Med12-PQL\n",
      "DBLINKS     NCBI-GeneID: 78174\n",
      "            NCBI-ProteinID: NP_084328\n",
      "            MGI: 1925424\n",
      "            Ensembl: ENSMUSG00000049387\n",
      "            Vega: OTTMUSG00000054260\n",
      "            UniProt: Q9D2H1\n",
      "AASEQ       82\n",
      "            MMFPLARYALNYLKTPSILKIVGRLKHSKPSSEETHDKYGNMMLISGTIFCLAGYTIYMT\n",
      "            QMGVEWNLSPIGRVTPQEWKKK\n",
      "NTSEQ       249\n",
      "            atgatgtttcccttggccagatatgcactaaactatctcaagactccaagcattctgaaa\n",
      "            attgtgggaagactgaaacactcaaagccctcatcagaagaaactcatgacaaatacggc\n",
      "            aatatgatgctaattagtgggactattttctgtttagctggatatacaatttatatgaca\n",
      "            cagatgggtgtggagtggaacctgtctcctattggcagagttaccccccaagagtggaag\n",
      "            aagaagtaa\n",
      "///\n",
      "\n",
      "['mmu00190  Oxidative phosphorylation']\n",
      "['mmu01100  Metabolic pathways']\n",
      "['mmu04260  Cardiac muscle contraction']\n",
      "['mmu04714  Thermogenesis']\n",
      "['mmu04932  Non-alcoholic fatty liver disease (NAFLD)']\n",
      "[\"mmu05010  Alzheimer's disease\"]\n",
      "[\"mmu05012  Parkinson's disease\"]\n",
      "[\"mmu05016  Huntington's disease\"]\n"
     ]
    },
    {
     "ename": "NameError",
     "evalue": "global name 'char' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-22-5c4d7385e43a>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m()\u001b[0m\n\u001b[0;32m     41\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     42\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 43\u001b[1;33m \u001b[0mletsrun\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     44\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     45\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m<ipython-input-22-5c4d7385e43a>\u001b[0m in \u001b[0;36mletsrun\u001b[1;34m()\u001b[0m\n\u001b[0;32m     29\u001b[0m     \u001b[1;32mwhile\u001b[0m \u001b[0mTrue\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     30\u001b[0m         \u001b[1;32mtry\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 31\u001b[1;33m             \u001b[0mrestart\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mchar\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mraw_input\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m\"Would you like to enter another [y/n] \"\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     32\u001b[0m             \u001b[1;32mif\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mrestart\u001b[0m \u001b[1;33m==\u001b[0m \u001b[1;34m'y'\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     33\u001b[0m                 \u001b[0mletsrun\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;31m#will rerun the program\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mNameError\u001b[0m: global name 'char' is not defined"
     ]
    }
   ],
   "source": [
    "# will output KEGG database and pathways\n",
    "def letsrun ():\n",
    "    \n",
    "    #READ IN SPECIES AND GENE\n",
    "    while True:\n",
    "        try:\n",
    "            species = raw_input(\"Enter a Species ( mouse is mmu)\")\n",
    "            gene = raw_input(\"Enter a Gene: \")\n",
    "\n",
    "            x = kegg_get(species+':' + gene).read()\n",
    "            break\n",
    "        except Exception:\n",
    "            print \"Oops!  That was not a valid gene/ and or species.  Try again...\"\n",
    "\n",
    "    #PRINT VALUES\n",
    "    print(kegg_get(species+':' + gene).read())\n",
    "    gene = REST.kegg_get(species+':'+gene).read()\n",
    "\n",
    "    for line in gene.rstrip().split(\"\\n\"):\n",
    "        #print \"here >> \", line, \"\\n\"\n",
    "        section = line[:12].strip()  # section names are within 12 columns\n",
    "        if not section == \"\":\n",
    "            current_section = section\n",
    "        \n",
    "        if current_section == \"PATHWAY\":\n",
    "            gene_identifiers= line[12:].split(\"; \")\n",
    "            print gene_identifiers\n",
    "    \n",
    "    while True:\n",
    "        try:\n",
    "            restart = raw_input(\"Would you like to enter another [y/n] \")\n",
    "            if(restart == 'y'):\n",
    "                letsrun() #will rerun the program\n",
    "            elif(restart == 'n'):\n",
    "                print \"All Done!\"\n",
    "            break\n",
    "        except ValueError:\n",
    "            print \"Oops!  That was not a valid entry. Please enter y or n\"\n",
    "            \n",
    "            \n",
    "   \n",
    "    \n",
    "letsrun()\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "#####Other mehtods of accessing database\n",
    "#print(kegg_info(\"kegg\").read())\n",
    "#print(kegg_info(\"pathway\").read())\n",
    "#print(kegg_find('mmu', 'cox7b').read())\n",
    "#print(kegg_find('genes', 'cd45').read())\n",
    "#print(kegg_list('mmu').read())\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'gene' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-13-a7cd14f06155>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m()\u001b[0m\n\u001b[0;32m      1\u001b[0m \u001b[1;31m#another way to parse data\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      2\u001b[0m \u001b[0mpathways\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;33m[\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m----> 3\u001b[1;33m \u001b[1;32mfor\u001b[0m \u001b[0mline\u001b[0m \u001b[1;32min\u001b[0m \u001b[0mgene\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mrstrip\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0msplit\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m\"\\n\"\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m      4\u001b[0m     \u001b[1;31m#print \"here >> \", line, \"\\n\"\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      5\u001b[0m     \u001b[0msection\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mline\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;36m12\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mstrip\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m  \u001b[1;31m# section names are within 12 columns\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mNameError\u001b[0m: name 'gene' is not defined"
     ]
    }
   ],
   "source": [
    "#another way to parse data\n",
    "pathways = []\n",
    "for line in gene.rstrip().split(\"\\n\"):\n",
    "    #print \"here >> \", line, \"\\n\"\n",
    "    section = line[:12].strip()  # section names are within 12 columns\n",
    "    if not section == \"\":\n",
    "            current_section = section\n",
    "        \n",
    "    if current_section == \"PATHWAY\":\n",
    "            gene_identifiers= line[12:].split(\" \")\n",
    "            #print gene_identifiers\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "restart = raw_input(\"Would you like to enter another [y/n] \")\n",
    "    if(restart == 'y'):\n",
    "        letsrun() #will rerun the program\n",
    "    elif(restart == 'n'):\n",
    "        print \"All Done!\"\n",
    "    else:\n",
    "        print \"please enter a correct command\"\n",
    "        \n",
    "          while True:\n",
    "        try:\n",
    "            restart = raw_input(\"Would you like to enter another [y/n] \")\n",
    "            if(restart == 'y'):\n",
    "            letsrun() #will rerun the program\n",
    "            elif(restart == 'n'):\n",
    "        print \"All Done!\n",
    "            break\n",
    "        except Exception:\n",
    "            print \"Oops!  That was not a valid entry. Please enter y or n\"\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.14"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
